Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java

```diff
@@ -85,6 +86,26 @@
         if (Objects.nonNull(process.getParent())) {
             addParentLinkForExport(prefs, workpiece, process.getParent());
         }
+
+        assignViewsFromChildrenRecursive(workpiece.getRootElement());
+    }
+
+    /**
+     * At all levels, assigns the views of the children to the included
+     * structural elements.
+     *
+     * @param includedStructuralElement
+     *            included structural element on which the recursion is
+     *            performed
+     */
+    private void assignViewsFromChildrenRecursive(IncludedStructuralElement includedStructuralElement) {
+        List<IncludedStructuralElement> children = includedStructuralElement.getChildren();
+        if (!children.isEmpty()) {
+            for (IncludedStructuralElement child : children) {
+                assignViewsFromChildrenRecursive(child);
+            }
+            MetadataEditor.assignViewsFromChildren(includedStructuralElement);
```

Comment:
this is also calling "assignViewsFromChildrenRecursive". I think there is something wrong here
Why are there two different methods for regular use and for recursive?

Reply:
This method recursively iterates through all children, calling `assignViewsFromChildren()` for each child. The method in the class `MetadataEditor` then assigns the views of all their children to the respective level. This must happen at all levels, not just at the top level. No duplicate references are generated, they are filtered out. I do not see where that should produce a wrong result.